.github/workflows/main.yml

```diff
@@ -0,0 +1,253 @@
+  pull_request:
+    branches:
+      - main
+  release:
+    types:
+      - published
+
+env:
+  # Setting these variables allows .NET CLI to use rich color codes in console output
+  TERM: xterm
+  DOTNET_SYSTEM_CONSOLE_ALLOW_ANSI_COLOR_REDIRECTION: true
+  # Skip boilerplate output
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+  DOTNET_NOLOGO: true
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+
+# Note that as much as we'd love to avoid repetitive work, splitting the pipeline into separate jobs
+# makes it very difficult to share artifacts between them. Even if we succeed, we'll still end up
+# pushing and pulling gigabytes worth of data, which makes the jobs so much slower that we might as
+# well just repeat the checkout-restore-build steps instead.
+
+# Having a setup that involves separate jobs gives us significant benefits, on the other hand, namely:
+# - Most of the jobs can run in parallel, which reduces the overall execution time significantly,
+#   despite the repetitive work.
+# - We can catch more issues this way, for example if the formatting job fails, we can still see the
+#   the test results too.
+# - If one of the jobs fails due to reasons unrelated to our code (e.g. NuGet server is down), we get
+#   the option to rerun only that job, saving us time.
+# - It's easier to understand what each job does (and later, read its output) because the scope is much
+#   more narrow.
+# - We can set permissions on a more granular (per-job) level, which allows us to expose only a few select
+#   steps to more sensitive access scopes.
+
+jobs:
+  # Check formatting
+  format:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Install .NET
+        uses: actions/setup-dotnet@607fce577a46308457984d59e4954e075820f10a # v3.0.3
+
+      - name: Validate format
+        run: dotnet format --verify-no-changes
+
+  # Run tests
+  test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Install .NET
+        uses: actions/setup-dotnet@607fce577a46308457984d59e4954e075820f10a # v3.0.3
+
+      - name: Run restore
+        run: dotnet restore
+
+      - name: Run build
+        run: >
+          dotnet build
+          --no-restore
+          --configuration Release
+
+      - name: Run tests
+        id: tests
+        run: >
+          dotnet test
+          --no-restore
+          --no-build
+          --configuration Release
+          --logger "trx;LogFileName=pw-test-results.trx"
+
+      - name: Report test results
+        uses: dorny/test-reporter@c9b3d0e2bd2a4e96aaf424dbaa31c46b42318226  # v1.6.0
+        # Only collect test results if the tests actually ran
+        if: ${{ steps.tests.outcome != 'cancelled' && steps.tests.outcome != 'skipped' }}
```

Comment:
This should prevent this step from failing when the tests step hasn't even been run (i.e. build or restore fails)